src/mpfao/3D/tdympfao3D_core.c

```diff
@@ -748,31 +759,34 @@ PetscErrorCode ComputeTransmissibilityMatrix_ForNonCornerVertex(TDy tdy,
       PetscInt isubcell = vertices->subcell_ids[vOffsetSubcell + i];
 
       PetscInt subcell_id = icell*cells->num_subcells[icell]+isubcell;
-      PetscInt sOffsetFace = subcells->face_offset[subcell_id];
 
-      PetscInt iface;
-      for (iface=0;iface<subcells->num_faces[subcell_id];iface++) {
+      PetscInt *subcell_face_ids, num_faces;
+      ierr = TDyMeshGetSubcellFaces(mesh, subcell_id, &subcell_face_ids, &num_faces); CHKERRQ(ierr);
+
+      for (PetscInt iface=0;iface<subcells->num_faces[subcell_id];iface++) {
 
-	PetscInt face_id = subcells->face_ids[sOffsetFace + iface];
-	PetscInt fOffsetCell = faces->cell_offset[face_id];
+        PetscInt face_id = subcell_face_ids[iface];
 
-	if (faces->is_internal[face_id] == 0) {
+        PetscInt *face_cell_ids, num_cell_ids;
+        ierr = TDyMeshGetFaceCells(mesh, face_id, &face_cell_ids, &num_cell_ids); CHKERRQ(ierr);
 
-	  // Extract pressure value at the boundary
-	  if (faces->cell_ids[fOffsetCell + 0] >= 0) idxBnd[numBnd] = -faces->cell_ids[fOffsetCell + 1] - 1;
-	  else                                       idxBnd[numBnd] = -faces->cell_ids[fOffsetCell + 0] - 1;
+        if (faces->is_internal[face_id] == 0) {
```

Comment:
Can also be spelled `if (!faces->is_internal[face_id]) {`, but that's a matter of taste.